src/components/tx/TxData.js

```diff
@@ -409,10 +448,12 @@ class TxData extends React.Component {
       )
     }
 
-    const renderGraph = (label, type) => {
+    const renderGraph = (label, type, toggleFunction, currentState) => {
       return (
-        <div className="mt-3 graph-div" id={`graph-${type}`} key={`graph-${type}-${this.props.transaction.hash}`}>
+        <div  id={`graph-neighbors`} key={`graph-${type}-${this.props.transaction.hash}`}>
           <label className="graph-label">{label}:</label>
+          <a href="true" className="ml-1" onClick={(e) => toggleFunction(e)}>{currentState ? 'Click to hide' : 'Click to show'}</a>
+          <div className={currentState ? undefined : 'hidden'} id={`graph-${type}`}></div>
```

Comment:
If the full node is loaded with requests it may take some time to load this graph and the user will see only an empty box. Maybe we could create another state `fundsGraphLoaded` (the same for verification) and inside this `div` we add a `{!fundsGraphLoaded && 'Loading graph...'}` so the user understands that something is loading. We can even add a Loading that is already in use in the project (`ReactLoading`)

What do you think?

Reply:
Great idea, I think this really improves the UX. Currently, if nothing happens for a while, the user will think the action did not work. He will try to reload the page, click again and generate more load on full-node.